controller/node_controller.go

```diff
@@ -357,27 +357,45 @@ func (nc *NodeController) syncDiskStatus(node *longhorn.Node) error {
 		}
 		// get disk available size
 		diskInfo, err := nc.getDiskInfoHandler(disk.Path)
+		readyCondition := types.GetDiskConditionFromStatus(diskStatus, types.DiskConditionTypeReady)
 		// if the file system has changed
 		if err != nil {
-			logrus.Errorf("Get disk information on node %v error: %v", node.Name, err)
+			if readyCondition.Status != types.ConditionStatusFalse {
+				readyCondition.LastTransitionTime = util.Now()
+			}
+			readyCondition.Status = types.ConditionStatusFalse
+			readyCondition.Reason = types.DiskConditionReasonDiskIsNotExist
+			readyCondition.Message = fmt.Sprintf("Get disk information on node %v error: %v", node.Name, err)
 			// disable invalid disk
 			updateDisk.AllowScheduling = false
 			updateDisk.StorageMaximum = 0
 			updateDisk.StorageReserved = 0
```

Comment:
Realized that you don't need to reset Reserved, which is a user setting and would be positive anyway.

Reply:
Seems we don't validate the user input to be positive.

Need to add that in a separate PR. Filed https://github.com/rancher/longhorn/issues/211